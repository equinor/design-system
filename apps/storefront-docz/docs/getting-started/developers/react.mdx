---
title: Developers
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
