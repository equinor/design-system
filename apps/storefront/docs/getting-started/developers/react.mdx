---
title: Developers
metaTitle: Developers react
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
