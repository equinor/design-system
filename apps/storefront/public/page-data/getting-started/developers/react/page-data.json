{"componentChunkName":"component---docs-getting-started-developers-react-mdx","path":"/getting-started/developers/react/","result":{"pageContext":{"frontmatter":{"title":"Developers","metaTitle":"Developers react","tabs":["Introduction","CSS framework","React","Figma"],"mode":"draft","route":"/getting-started/developers/react/"},"entry":{"id":"6f9784da00517d57e4627269567073e6","filepath":"docs/getting-started/developers/react.mdx","fullpath":"/Users/vnys/Kode/storefront-replace/apps/storefront/docs/getting-started/developers/react.mdx","route":"/getting-started/developers/react/","slug":"docs-getting-started-developers-react","name":"React","menu":"","headings":[]}}}}