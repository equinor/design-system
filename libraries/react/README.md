@eds-ui/react
===============

1. `npm i -g pnpm`
1. `pnpm install`
1. `pnpm recursive install`

## Run storybook

`pnpm start`

## Build modules

`pnpm recursive run build`
