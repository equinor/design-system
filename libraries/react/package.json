{
  "name": "@eds-ui/react",
  "private": true,
  "version": "0.0.1",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "jest",
    "storybook": "start-storybook"
  },
  "keywords": [],
  "author": "",
  "license": "AGPL",
  "dependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.4",
    "@babel/preset-env": "^7.4.4",
    "@babel/preset-react": "^7.0.0",
    "@storybook/react": "^5.0.11",
    "babel-plugin-styled-components": "^1.10.0",
    "enzyme": "^3.9.0",
    "enzyme-adapter-react-16": "^1.12.1",
    "jest": "^24.8.0",
    "jest-styled-components": "^6.3.1",
    "rollup": "^1.11.3",
    "rollup-plugin-babel": "^4.3.2",
    "rollup-plugin-commonjs": "^9.3.4",
    "rollup-plugin-node-resolve": "^4.2.3",
    "styled-components": "^4.2.0"
  },
  "peerDependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "styled-components": "^4.2.0",
    "prop-types": "^15.7.2"
  },
  "jest": {
    "verbose": true,
    "setupFilesAfterEnv": [
      "<rootDir>/testSetup.js"
    ],
    "testPathIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "transformIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "haste": {
      "providesModuleNodeModules": [
        ".*"
      ]
    },
    "roots": [
      "packages/node_modules"
    ]
  }
}
