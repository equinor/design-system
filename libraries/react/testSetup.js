import React from 'react'
import Enzyme from 'enzyme'
import 'jest-styled-components'
import styled from 'styled-components'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })
