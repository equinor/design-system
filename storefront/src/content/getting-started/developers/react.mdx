---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
mode: draft
---


:warning: Information coming soon
