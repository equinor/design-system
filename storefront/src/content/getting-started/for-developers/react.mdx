---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
---

Developers React
