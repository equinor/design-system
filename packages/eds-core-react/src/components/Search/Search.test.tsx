import { render, fireEvent, screen, act } from '@testing-library/react'
import { axe } from 'jest-axe'
import styled from 'styled-components'
import { Search } from '.'

const StyledSearch = styled(Search)`
  position: relative;
  width: 50%;
`

describe('Search', () => {
  it('Matches snapshot', () => {
    const { asFragment } = render(<Search />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('Should pass a11y test', async () => {
    /** workaround the console error */
    // eslint-disable-next-line @typescript-eslint/unbound-method
    const { getComputedStyle } = window
    window.getComputedStyle = (elt) => getComputedStyle(elt)

    const { container } = render(
      <Search aria-label="Expect user to declare label" />,
    )
    await act(async () => {
      const result = await axe(container)
      expect(result).toHaveNoViolations()
    })
  })
  it('Can extend the css for the component', () => {
    render(<StyledSearch data-testid="search" />)
    const search = screen.getByRole('search')
    expect(search).toHaveStyleRule('position', 'relative')
    expect(search).toHaveStyleRule('width', '50%')
  })
  it('Has rendered provided value in input field', () => {
    const value = 'provided value'

    render(<Search value={value} data-testid="search" />)

    const searchInput = screen.getByTestId('search')

    expect(searchInput).toHaveValue(value)
  })
  it('Has called onChange once with event & new value, when value is changed', () => {
    const searchId = 'search-id-when-testing'
    const newValue = 'some new value'
    let callbackId = ''
    let callbackValue = ''
    const handleOnChange = jest.fn(({ target: { id, value } }) => {
      callbackId = id as string
      callbackValue = value as string
    })

    render(
      <Search
        id={searchId}
        value="some old value"
        onChange={handleOnChange}
        data-testid="search"
      />,
    )
    const searchInput = screen.queryByTestId('search')

    fireEvent.change(searchInput, {
      target: { value: newValue },
    })

    expect(handleOnChange).toHaveBeenCalled()
    expect(callbackId).toEqual(searchId)
    expect(callbackValue).toEqual(newValue)
  })

  it('Has cleared value when clear button is clicked', () => {
    const searchId = 'search-id-when-testing'
    let callbackId = ''
    let callbackValue = ''
    const handleOnChange = jest.fn(({ target: { id, value } }) => {
      callbackValue = value as string
      callbackId = id as string
    })

    render(
      <Search
        id={searchId}
        defaultValue="initial value"
        onChange={handleOnChange}
        data-testid="search"
      />,
    )
    const clearButton = screen.queryByRole('button')
    const searchInput = screen.queryByTestId('search')
    fireEvent.click(clearButton)

    expect(handleOnChange).toHaveBeenCalled()
    expect(searchInput).toHaveValue('')
    expect(callbackValue).toEqual('')
    expect(callbackId).toEqual(searchId)
  })

  it('Has called onFocus when search if focused', () => {
    const searchId = 'search-id-when-testing'
    let callbackId = ''
    const handleOnFocus = jest.fn(({ target: { id } }) => {
      callbackId = id as string
    })

    render(
      <Search id={searchId} onFocus={handleOnFocus} data-testid="search" />,
    )
    const searchInput = screen.queryByTestId('search')

    fireEvent.focus(searchInput)

    expect(handleOnFocus).toHaveBeenCalled()
    expect(callbackId).toEqual(searchId)
  })
  it('Has called onBlur when search if blured', () => {
    const searchId = 'search-id-when-testing'
    let callbackId = ''
    const handleOnBlur = jest.fn(({ target: { id } }) => {
      callbackId = id as string
    })

    render(<Search id={searchId} onBlur={handleOnBlur} data-testid="search" />)
    const searchInput = screen.queryByTestId('search')

    fireEvent.blur(searchInput)

    expect(handleOnBlur).toHaveBeenCalled()
    expect(callbackId).toEqual(searchId)
  })

  it('Has new value, when value property is changed after first render', () => {
    const { rerender } = render(<Search value="old" data-testid="search-old" />)

    rerender(<Search value="new" data-testid="search-new" />)

    const searchInput: HTMLInputElement = screen.queryByTestId('search-new')

    expect(searchInput.value).toEqual('new')
  })
})
