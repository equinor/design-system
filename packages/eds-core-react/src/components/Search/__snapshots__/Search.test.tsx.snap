// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  --eds-input-adornment-color: var(--eds_text__static_icons__tertiary,rgba(111,111,111,1));
  --eds-input-color: var(--eds_text__static_icons__default,rgba(61,61,61,1));
  position: relative;
  height: 36px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border: none;
  box-sizing: border-box;
  box-shadow: inset 0px -1px 0px 0px var(--eds_text__static_icons__tertiary,rgba(111,111,111,1));
  background: var(--eds-input-background,var(--eds_ui_background__light,rgba(247,247,247,1)));
  outline: 1px solid transparent;
  outline-offset: 0px;
}

.c0:focus-within {
  box-shadow: none;
  outline: 2px solid var(--eds_interactive_primary__resting,rgba(0,112,121,1));
  outline-offset: 0px;
}

.c0 > input {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c4 {
  width: 100%;
  border: none;
  background: transparent;
  padding-left: 40px;
  padding-top: 6px;
  padding-right: 40px;
  padding-bottom: 6px;
  font-family: Equinor;
  font-size: 1.000rem;
  font-weight: 400;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  line-height: 1.500em;
  outline: none;
  padding-left: 40px;
  padding-right: 40px;
}

.c4::-webkit-input-placeholder {
  color: var(--eds_text__static_icons__tertiary,rgba(111,111,111,1));
}

.c4::-moz-placeholder {
  color: var(--eds_text__static_icons__tertiary,rgba(111,111,111,1));
}

.c4:-ms-input-placeholder {
  color: var(--eds_text__static_icons__tertiary,rgba(111,111,111,1));
}

.c4::placeholder {
  color: var(--eds_text__static_icons__tertiary,rgba(111,111,111,1));
}

.c4:disabled {
  color: var(--eds-input-color);
  cursor: not-allowed;
}

.c2 {
  position: absolute;
  top: 6px;
  bottom: 6px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Equinor;
  font-size: 0.750rem;
  font-weight: 500;
  line-height: 1.333em;
  color: var(--eds-input-adornment-color);
}

.c3 {
  left: 0;
  padding-left: 8px;
}

.c5 {
  right: 0;
  padding-right: 8px;
}

.c1 input[type='search']::-webkit-search-decoration,.c1 input[type='search']::-webkit-search-cancel-button,.c1 input[type='search']::-webkit-search-results-button,.c1 input[type='search']::-webkit-search-results-decoration {
  -webkit-appearance: none;
}

<div
    class=""
    role="search"
  >
    <div
      class="c0 c1"
    >
      <div
        class="c2 c3"
      >
        <svg
          aria-hidden="true"
          class=""
          fill="currentColor"
          height="18px"
          viewBox="0 0 18 18"
          width="18px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class=""
            clip-rule="evenodd"
            d="M11.224 10.691h.592l3.743 3.75-1.118 1.118-3.75-3.743v-.592l-.202-.21a4.853 4.853 0 0 1-3.173 1.177 4.875 4.875 0 1 1 4.875-4.875 4.853 4.853 0 0 1-1.177 3.173l.21.202zM3.94 7.316a3.37 3.37 0 0 0 3.375 3.375 3.37 3.37 0 0 0 3.375-3.375 3.37 3.37 0 0 0-3.375-3.375 3.37 3.37 0 0 0-3.375 3.375z"
            data-testid="eds-icon-path"
            fill-rule="evenodd"
            height="18"
          />
        </svg>
      </div>
      <input
        class="c4"
        style="resize: none;"
        type="text"
      />
      <div
        class="c2 c5"
      />
    </div>
  </div>
</DocumentFragment>
`;
